src/libtools/examples/merging.cpp

```diff
@@ -0,0 +1,173 @@
+	// it seems at first. Have a look at the documentation of kdbGet
+	// for detailed information
+	// things to note:
+	//   * use blocks with local KDB instances so we don't have to worry about
+	//     writing the keys back
+	//   * remove the root key itself from the result KeySet because it usually
+	//     contains the mounted filename and cannot be merged anyway
+	{
+		KDB lkdb;
+		lkdb.get (ours, oursRoot);
+		ours = ours.cut (oursRoot);
+		ours.lookup(oursRoot, KDB_O_POP);
+	}
+	{
+		KDB lkdb;
+		lkdb.get (theirs, theirsRoot);
+		theirs = theirs.cut (theirsRoot);
+		ours.lookup(oursRoot, KDB_O_POP);
+	}
+	{
+		KDB lkdb;
+		lkdb.get (base, baseRoot);
+		base = base.cut (baseRoot);
+		ours.lookup(oursRoot, KDB_O_POP);
+	}
+
+
+	// Step 2: Make sure that no keys reside below the intended merge result root
+	// Usually the merge can be either aborted if this is the case or the existing
+	// keys can be overwritten.
+	KeySet resultKeys;
+	kdb::KDB kdb;
+	kdb.get (resultKeys, resultRoot);
+
+	KeySet discard = resultKeys.cut (resultRoot);
+	if (discard.size () != 0)
+	{
+		// handle existing keys below the result root
+		return -1;
+	}
+
+	ThreeWayMerge merger;
+
+	// Step 3: Decide which resolution strategies to use. The stratgies are registered
+	// with the merge and applied in order as soon as a conflict is detected. If a strategy
+	// marks a conflict as resolved, no further strategies are consulted. Therefore the order
+	// in which they are registered is absolutely crucial. With this chaining the strategies
+	// remain simple, but can be combined to powerful resolution strategies.
+	// Have a look at the strategy documentation for further details on what they do and how they work.
+	// The unit tests also provide insight into how the strategies work.
+
+	// TODO:
+	// Currently the MetaMergeStrategy has to be registered first. Otherwise the meta information
+	// of merged keys is lost. This is due to the way that strategies are currently implemented and
+	// is subject to change.
+
+	// TODO:
+	// Currently the strategies have to be instantiated and freed manually. This is also subject to change
+	// and will be simplified in future versions.
+
+
+
+	MetaMergeStrategy metaStrategy(merger);
+	merger.addConflictStrategy(&metaStrategy);
+
+	// in this example we first resolve all the keys that can be automatically
+	// resolved (e.g. only one side was modified). This is done by the AutoMergeStrategy.
+	// If keys remain that cannot be automatically merged (e.g. both sides were modified)
+	// we unconditionally accept our version of the key. This is done by the OneSideStrategy.
+	// MetaKeys are merged in the exact same way, because the MetaMergeStrategy we instantiated
+	// above uses the same merger instance. Alternatively we could have instantiated and configured
+	// a different merger only for the MetaKeys.
+
+	MergeConflictStrategy *autoMerge = new AutoMergeStrategy();
+	MergeConflictStrategy *ourVersion = new OneSideStrategy(OURS);
```

Comment:
Mmh, I still have the feeling that OURS is really generic for the global namespace..